core/src/main/scripts/importer/validateData.py

```diff
@@ -990,9 +990,10 @@ def parseFeatureColumns(self, nonsample_col_vals):
             # treat empty string as a missing value
             if hugo_symbol == '':
                 hugo_symbol = None
-            # In case of CNA data the Hugo Symbol should be split when gene symbol contains pipe
-            if (type(self) is CNAValidator or type(self) is CNAContinuousValuesValidator) and '|' in hugo_symbol:
-                hugo_symbol = hugo_symbol.split('|')[0]
+            if hugo_symbol is not None:
+                # In case of CNA data the Hugo Symbol should be split when gene symbol contains pipe
+                if (type(self) is CNAValidator or type(self) is CNAContinuousValuesValidator) and '|' in hugo_symbol:
```

Comment:
You shouldn't have to modify a function that affects parsing of all possible gene-wise matrices in order to change something specific to CNA files. Would it beat all possible to move this logic to in a subclass?

Reply:
Having the superclass know about its subclasses in addition to vice-versa makes the code very hard to read.